stock_location_flowable/models/stock_location.py

```diff
@@ -0,0 +1,219 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import json
+
+
+class Location(models.Model):
+    _inherit = "stock.location"
+
+    flowable_storage = fields.Boolean()
+    flowable_mix_blocked = fields.Boolean(compute="_compute_flowable_mix_blocked")
+    flowable_capacity = fields.Float()
+    flowable_uom_id = fields.Many2one(comodel_name="uom.uom", string="Unit of Measure")
+    flowable_sequence_id = fields.Many2one(comodel_name="ir.sequence")
+    flowable_allowed_product_ids = fields.Many2many(
+        comodel_name="product.product", string="Allowed Products"
+    )
+    flowable_mixing_production_id = fields.Many2one(
+        comodel_name="mrp.production",
+        compute="_compute_flowable_mixing_production_id",
+        store=True,
+    )
+    flowable_capacity_occupied = fields.Float(
+        compute="_compute_flowable_capacity_occupied"
+    )
+    # Mirar el tema de perque la string esta " " mira el percentpie
+    flowable_percentage_occupied = fields.Float(
+        compute="_compute_flowable_percentage_occupied"
+    )
+    use_new_order_lot = fields.Boolean()
+    json_popover = fields.Char(
+        "JSON data for the popover widget", compute="_compute_json_popover"
+    )
+
+    def _compute_json_popover(self):
+        for rec in self:
+            rec.json_popover = json.dumps(
+                {
+                    "title": _("Flowable Location Warning"),
+                    "msg": _(
+                        "This location is blocked because it is a flowable"
+                        " and it has a manufacturing order assigned."
+                    ),
+                }
+            )
+
+    @api.depends("flowable_mixing_production_id")
+    def _compute_flowable_mix_blocked(self):
+        for rec in self:
+            rec.flowable_mix_blocked = (
+                True
+                if rec.flowable_storage and rec.flowable_mixing_production_id
+                else False
+            )
+
+    @api.depends(
+        "usage",
+        "flowable_storage",
+        "quant_ids.product_id.stock_move_ids.move_line_ids.state",
```

Comment:
Maybe instead of navigating `quants`, could it be better to use a `One2Many` field to get directly to the` stock.move.line` (as a `location_dest_id` and `location_id`) related to this location 
Maybe we should use the parameter check_company=True in the field??. In any case, check in all the code if we can use that parameter to simplify the multi-company scenario.